tests/tests.cmake

```diff
@@ -98,13 +98,13 @@ function(add_generic_test)
         set(test_name "${name}-${board}")
 
         if(DEFINED generated_xdc)
-            set(xdc ${generated_xdc})
+            set(xdc ${CMAKE_CURRENT_SOURCE_DIR}/${generated_xdc})
```

Comment:
I think this change might break some litex tests

Reply:
Ok, I can fix that. Aren't those tests a part of the CI? It passed with this change.

Reply:
Right, looking at the log there actually was a failure related to this, as the XDC was generated by LiteX and it is not present in the `CURRENT_SOURCE_DIR`. 

```
make[3]: *** No rule to make target '../tests/designs/litex_base/tmpfs/src/github/fpga-interchange-tests/build/tests/designs/litex_base/digilent_arty/gateware/digilent_arty.xdc', needed by 'tests/designs/litex_base/arty35t/litex_base.vivado.bit'.
```

Possibly the test was set as `failure_allowed` probably due to some issues with the bitstream generation.

Reply:
Ah, right. Ok, I'll update the mechanism of providing design-specific XDC

Reply:
Actually, the mechanism already exists. If the XDC is generated, by setting the `generated_xdc` will override the default XDC location.

If no XDC argument is passed, than the default location of the XDC is in the test's directory and the name of the XDC needs to be `<board_name>.xdc`

Reply:
hm, actually right - I can set `generated_xdc` to `${CURRENT_SOURCE_DIR}/something` and it should work

Reply:
I believe that you can just skip all together the assignment of the XDC, and just name [this file](https://github.com/SymbiFlow/fpga-interchange-tests/blob/9703fdba1dd9441eb2d4af29b5e69cebe13b6046/tests/features/lut/lut_routethru.xdc) `testboard_noffmux.xdc` rather than `lut_routethru.xdc`

Reply:
But there are multiple LUT tests in the same folder and they have different top-level IO counts. So in this case the name not only depends on board but also on test.

Reply:
Yeah, but the test defined in CMake targets only one of the boards, so the routethrough test will target only the noffmux testboard, therefore only that XDC will be taken into account.

Reply:
True, but `lut_ff` also targets the same board but the xdc needs to be different due to different top-level IOs

Reply:
Aha all right, I misread that and didn't see that `lut_ff` now has both the testboards